datastore/bolt/repo.go

```diff
@@ -132,23 +132,43 @@ func (db *DB) SetRepoKeypair(repo string, keypair *common.Keypair) error {
 
 // DelRepo deletes the repository.
 func (db *DB) DelRepo(repo *common.Repo) error {
-	//TODO(benschumacher) rework this to use BoltDB's txn wrapper
-
-	t, err := db.Begin(true)
-	if err != nil {
-		return err
-	}
 	key := []byte(repo.FullName)
-	err = t.Bucket(bucketRepo).Delete(key)
-	if err != nil {
-		t.Rollback()
+
+	return db.Update(func(t *bolt.Tx) error {
+		err := t.Bucket(bucketRepo).Delete(key)
+		if err != nil {
+			return err
+		}
+		t.Bucket(bucketRepoKeys).Delete(key)
+		t.Bucket(bucketRepoParams).Delete(key)
+
+		// should we just ignore these error conditions? or should
```

Comment:
Whoops. I meant to clean this up a bit more, but how should we handle errors. In practice, I think BoltDB is unlikely to fail through this code path, but I wasn't sure if it was reasonable to rollback the entire "delete" operation if one part hiccuped.